src/main/java/de/tum/in/www1/artemis/domain/AuxiliaryRepository.java

```diff
@@ -0,0 +1,176 @@
+        return checkoutDirectory;
+    }
+
+    public void setCheckoutDirectory(String checkoutDirectory) {
+        this.checkoutDirectory = checkoutDirectory;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public ProgrammingExercise getExercise() {
+        return exercise;
+    }
+
+    public void setExercise(ProgrammingExercise exercise) {
+        this.exercise = exercise;
+    }
+
+    /**
+     * Returns the fully qualified name of the repository, which looks like
+     * [exercise identifier]-[repository name]
+     *
+     * @return Fully qualified name of the repository
+     */
+    @JsonIgnore
+    public String getRepositoryName() {
+        return exercise.generateRepositoryName(getName());
+    }
+
+    /**
+     * Returns whether this auxiliary repository should be included in the exercise's build plan or not.
+     * The repository is included when the repository url and the checkout directory have appropriate
+     * values.
+     *
+     * @return true or false whether this repository should be included in the exercise build plan or not
+     */
+    @JsonIgnore
+    public boolean shouldBeIncludedInBuildPlan() {
+        return getCheckoutDirectory() != null && !getCheckoutDirectory().isBlank() && getRepositoryUrl() != null && !getRepositoryUrl().isBlank();
+    }
+
+    /**
+     * Returns a copy of this auxiliary repository object without including the repository url.
+     * This is used when importing auxiliary repositories from an old exercise to a new exercise.
+     *
+     * @return A clone of this auxiliary repository object except the repository url
+     */
+    @JsonIgnore
+    public AuxiliaryRepository cloneObjectForNewExercise() {
+        AuxiliaryRepository newAuxiliaryRepository = new AuxiliaryRepository();
+        newAuxiliaryRepository.name = this.name;
+        newAuxiliaryRepository.checkoutDirectory = this.checkoutDirectory;
+        newAuxiliaryRepository.description = this.description;
+        return newAuxiliaryRepository;
+    }
+
+    /**
+     * Gets a URL of the repositoryUrl if there is one
+     *
+     * @return a URL object of the repositoryUrl or null if there is no repositoryUrl
+     */
+    @JsonIgnore
+    public VcsRepositoryUrl getVcsRepositoryUrl() {
+        String repositoryUrl = getRepositoryUrl();
+        if (repositoryUrl == null || repositoryUrl.isEmpty()) {
+            return null;
+        }
+        try {
+            return new VcsRepositoryUrl(repositoryUrl);
+        }
+        catch (MalformedURLException e) {
+            LoggerFactory.getLogger(AuxiliaryRepository.class).error("Malformed URL " + getRepositoryUrl() + " you could not be used to instantiate VcsRepositoryUrl.", e);
```

Comment:
Could be moved to a constant.

Reply:
Fixed in https://github.com/ls1intum/Artemis/pull/3457/commits/415d2021c642b223e9aa038d860cd48e3f706801